driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -543,4 +591,59 @@ public String toString() {
                 + ", mechanismProperties=<hidden>"
                 + '}';
     }
+
+    /**
+     * The context for the {@link OidcRequestCallback#onRequest(OidcRequestContext) OIDC request callback}.
+     */
+    @Evolving
+    public interface OidcRequestContext {
+
+        /**
+         * @return The timeout that this callback must complete within.
+         */
+        Duration getTimeout();
+
+        /**
+         * @return The OIDC callback API version. Currently, version 1.
+         */
+        int getVersion();
+    }
+
+    /**
+     * This callback is invoked when the OIDC-based authenticator requests
+     * tokens from the identity provider.
+     * <p>
+     * It does not have to be thread-safe, unless it is provided to multiple
+     * MongoClients.
+     */
+    public interface OidcRequestCallback {
+        /**
+         * @param context The context.
+         * @return The response produced by an OIDC Identity Provider
+         */
+        RequestCallbackResult onRequest(OidcRequestContext context);
+    }
+
+    /**
+     * The response produced by an OIDC Identity Provider.
+     */
+    public static final class RequestCallbackResult {
+
+        private final String accessToken;
```

Comment:
Make sure that `expiresIn` (expressed as `Duration`?) accompanies the `accessToken` field.

Reply:
This is accepted for the sake of the future (as per the spec), but not actually used in the driver/spec.